### PR TITLE
Release/3.2.0

### PR DESCRIPTION
**Pull request by MouazAlzahabi**

### CODE REVIEW

File: Multisig/UI/SwiftUI/UI Library/Overlays/Snackbar/SnackbarCenter.swift [hunk 1]
```diff
@@ -60,9 +60,9 @@ class SnackbarCenter: ObservableObject {
         }
     }
 
-    func show(error: DetailedLocalizedError) {
+    func show(error: Error) {
```
Comment (sche) [inline Multisig/UI/SwiftUI/UI Library/Overlays/Snackbar/SnackbarCenter.swift]: @MouazAlzahabi  I don't remember exactly, but I thought you removed it in your previous PR. Should it be like this?

File: Multisig/UI/Transaction/TransactionDetailsViewController/RejectionConfirmationViewController/RejectionConfirmationViewController.swift [hunk 2]
```diff
@@ -218,7 +217,9 @@ extension RejectionConfirmationViewController: SelectLedgerDeviceDelegate {
 
         let pendingConfirmationVC = LedgerPendingConfirmationViewController(ledgerHash: rejectionTransaction.hardwareWalletHash)
         pendingConfirmationVC.modalPresentationStyle = .popover
-        pendingConfirmationVC.onClose = {
+        pendingConfirmationVC.onClose = { [weak self] in
+            self?.ledgerController = nil
+            self?.endLoading()
             controller.reloadData()
```
Comment (sche) [inline Multisig/UI/Transaction/TransactionDetailsViewController/RejectionConfirmationViewController/RejectionConfirmationViewController.swift]: `controller.reloadData()` should be removed

File: Multisig/UI/Transaction/TransactionDetailsViewController/TransactionDetailsViewController.swift [hunk 3]
```diff
@@ -522,7 +522,9 @@ extension TransactionDetailsViewController: SelectLedgerDeviceDelegate {
 
         let pendingConfirmationVC = LedgerPendingConfirmationViewController(ledgerHash: transaction.hardwareWalletHash)
         pendingConfirmationVC.modalPresentationStyle = .popover
-        pendingConfirmationVC.onClose = {
+        pendingConfirmationVC.onClose = { [weak self] in
+            self?.ledgerController = nil
+            self?.reloadData()
             controller.reloadData()
```
Comment (sche) [inline Multisig/UI/Transaction/TransactionDetailsViewController/TransactionDetailsViewController.swift]: `controller.reloadData()` should be removed

File: Multisig/UI/SwiftUI/UI Library/Overlays/Snackbar/SnackbarCenter.swift (hunk 1 above)
Comment (MouazAlzahabi) [reply Multisig/UI/SwiftUI/UI Library/Overlays/Snackbar/SnackbarCenter.swift]: @DmitryBespalov this related to one of your PRs right ?

Comment (DmitryBespalov) [reply Multisig/UI/SwiftUI/UI Library/Overlays/Snackbar/SnackbarCenter.swift]: yes, because of it we had crash in forced unwrapping

**Review by DmitryBespalov — APPROVED** [review]

File: Multisig/UI/SwiftUI/UI Library/Overlays/Snackbar/SnackbarCenter.swift (hunk 1 above)
Comment (sche) [reply Multisig/UI/SwiftUI/UI Library/Overlays/Snackbar/SnackbarCenter.swift]: but then it breaks our requirement of how error messages should look like. Is there another way to fix that crash?

Comment (DmitryBespalov) [reply Multisig/UI/SwiftUI/UI Library/Overlays/Snackbar/SnackbarCenter.swift]: tbh not sure which error is causing the trouble 😞

Comment (DmitryBespalov) [reply Multisig/UI/SwiftUI/UI Library/Overlays/Snackbar/SnackbarCenter.swift]: @sche here's the PR https://github.com/gnosis/safe-ios/pull/1469/files

Comment (sche) [reply Multisig/UI/SwiftUI/UI Library/Overlays/Snackbar/SnackbarCenter.swift]: In general, I am fine with it if you are fine to remove this hard requirement of displaying messages to users only that satisfy the DetailedLocalizedError format.

Comment (MouazAlzahabi) [reply Multisig/UI/SwiftUI/UI Library/Overlays/Snackbar/SnackbarCenter.swift]: @DmitryBespalov @sche So let's go with it ?

Comment (sche) [reply Multisig/UI/SwiftUI/UI Library/Overlays/Snackbar/SnackbarCenter.swift]: Yes, you can merge it @MouazAlzahabi 
We will adjust it later if needed

**Review by sche — APPROVED** [review]

**Pull request merged**